pkg/keyidentifier/testdata/make.sh

```diff
@@ -3,97 +3,165 @@
-    mkdir -p {encrypted,plaintext}"/$dir_fragment"
-    ssh-keygen -t $type -b $bits  -C "" -f "plaintext/$dir_fragment/seph-$os" -P ""
-    ssh-keygen -t $type -b $bits  -C "" -f "encrypted/$dir_fragment/seph-$os" -P "$(rand)"
-}
-
-makeOpensshKey rsa 1024
-makeOpensshKey rsa 2048
-makeOpensshKey rsa 4096
-
-makeOpensshKey dsa 1024
-
-makeOpensshKey ecdsa 256
-makeOpensshKey ecdsa 521
+    encrypted=$3
+    format="openssh7"
+    source="openssh"
+
+    keypath="$DATA_DIR/$(rand)"
+
+    if [ $encrypted == true ]; then
+        ssh-keygen -t $type -b $bits  -C "" -f $keypath -P "$(rand)"
+        cmd='ssh-keygen -t $type -b $bits  -C \"\" -f $keypath -P $(rand)'
+    else
+        ssh-keygen -t $type -b $bits  -C "" -f $keypath -P ""
+        cmd='ssh-keygen -t $type -b $bits  -C \"\" -f $keypath -P \"\"'
+    fi
 
-# ed25519 is fixed legth, bits is ignored
-makeOpensshKey ed25519 2048
+    fingerprint=$(ssh-keygen -l -f $keypath | awk '{print $2}')
+    md5fingerprint=$(ssh-keygen -l -E md5 -f $keypath | awk '{print $2}' | sed 's/^MD5://')
+
+    cat <<EOF > $keypath.json
+    {
+      "FingerprintSHA256": "$fingerprint",
+      "FingerprintMD5": "$md5fingerprint",
+      "Type": "$type",
+      "Bits": $bits,
+      "Encrypted": $encrypted,
+      "command": "$cmd",
+      "Format": "openssh-new",
+      "Source": "ssh-keygen"
+    }
+EOF
+}
 
 # ssh.com style
 #
 # Note that openssh's ssh-keygen proportes to convert (using `-e`) but
 # empirically this does not work. So we use puttygen (`brew install
 # putty` to generate these)
 
-function makePuttyKeyPuttyFormat {
+# check if puttygen is installed
+hash puttygen 2>/dev/null || \
+    { echo >&2 "puttygen must be installed to generate test data for putty keys. use 'brew install putty' to install puttygen on macos"; exit 1; }
+
+function makePuttyKeyAndSpecFile {
     type=$1
     bits=$2
-    format="putty"
-
-    if [ "$type" == "rsa1" ]; then
-        format="ssh1"
+    format=$3
+    encrypted=$4
+    source="putty"
+
+    if [ "$format" == "putty" ]; then
+        putty_format=""
+        if [ "$type" == "rsa1" ]; then
+            format="ssh1"
+        fi
+    else
+        putty_format="-$format"
     fi
 
-    dir_fragment="$type/$bits/$format"
-    mkdir -p {encrypted,plaintext}"/$dir_fragment"
+    keypath="$DATA_DIR/$(rand)"
 
-    puttygen -t $type -b $bits -C "" -o "plaintext/$dir_fragment/seph-putty" -O private --new-passphrase /dev/null
-    puttygen -t $type -b $bits -C "" -o "encrypted/$dir_fragment/seph-putty" -O private --new-passphrase <(rand)
+    passphrase=""
+    if [ $encrypted == true ]; then
+        passphrase=$(rand)
+        puttygen -t $type -b $bits -C "" -o $keypath -O private$putty_format --new-passphrase <(echo $passphrase)
+        cmd="puttygen -t $type -b $bits -C '' -o $keypath -O private$putty_format --new-passphrase $passphrase"
+    else
+        puttygen -t $type -b $bits -C "" -o $keypath -O private$putty_format --new-passphrase <(cat /dev/null)
```

Comment:
```suggestion
        puttygen -t $type -b $bits -C "" -o $keypath -O private$putty_format --new-passphrase /dev/null
```

`<(cat /dev/null)` is redundant. just `/dev/null`. The `<(...)` is making a file handle out of stdout. `/dev/null` is already a file.

Reply:
ah right, I see. This is rendered moot w/the commit I have locally